ssl/ssl_sess.c

```diff
@@ -361,6 +361,11 @@ unsigned int SSL_SESSION_get_compress_id(const SSL_SESSION *s)
     return s->compress_meth;
 }
 
+const char *SSL_SESSION_get_hostname(const SSL_SESSION *s)
+{
+    return s->tlsext_hostname;
+}
```

Comment:
If you want to return a pointer to the internal object, we seem to be using "get0" as function name.